File: app/src/main/java/com/walmartlabstest/walmartproducts/uiutils/StringUtils.java
package com.walmartlabstest.walmartproducts.uiutils;

public class StringUtils {

    public static final String products="products";

    private static StringUtils instance = null;
    public static String position ="positionSelected";

    public static StringUtils getInstance(){

        if(instance == null){

            instance = new StringUtils();
        }
        return instance;
    }

    private StringUtils(){

    }

    public boolean isNullEmpty(String text){

        if(text==null || text.isEmpty()){
            return true;
        }
        return false;
    }
}

File: app/src/main/java/com/walmartlabstest/walmartproducts/uiutils/StringUtils.kt
package com.walmartlabstest.walmartproducts.uiutils

class StringUtils private constructor() {
    fun isNullEmpty(text: String?): Boolean {
        return if (text == null || text.isEmpty()) {
            true
        } else false
    }

    companion object {
        const val products = "products"
        var instance: StringUtils? = null
            get() {
                if (field == null) {
                    field = StringUtils()
                }
                return field
            }
            private set
        var position = "positionSelected"
    }
}